src/provider/model/APIHandler.ts

```diff
@@ -102,11 +106,16 @@ export class APIHandler<T extends Enum, E extends EventSpecification> {
     }
 
     // TODO?: Remove the need for this any by defining connection payload type?
-    private onConnection(app: Identity, payload?: any): void {
+    private onConnectionHandler(app: Identity, payload?: any): void {
         if (payload && payload.version && payload.version.length > 0) {
             console.log(`connection from client: ${app.name}, version: ${payload.version}`);
         } else {
             console.log(`connection from client: ${app.name}, unable to determine version`);
         }
+        this.onConnection.emit(app);
```

Comment:
Did this work for you? When doing something similar on FDC3 we noticed that this event fires just before the window _actually_ connects, meaning that you can't communicate with the window from this signal.

See here: https://github.com/HadoukenIO/fdc3-service/blob/develop/src/provider/APIHandler.ts#L117-L121

Reply:
never had any issues regarding this in my tests, but could be timing-related. I'm applying the fix just in case.